src/hooks/ts/useSearchParams.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react"
+
+type TUseSearchParams = ( url?: string, opt?: { unique: boolean } ) => Record<string, any>
+export const useSearchParams: TUseSearchParams = ( url = location.search, opt = { unique: true } ) => {
+  const _urlSearch = new URL(url)
```

Comment:
When I use the hook without specify an url this error ocurrs:

useSearchParams.ts:5 Uncaught TypeError: Failed to construct 'URL': Invalid URL at useSearchParams (useSearchParams.ts:5:22)

```ts
import { useSearchParams } from "./hooks/ts/useSearchParams";

const AppTs = () => {
  const params = useSearchParams();
  return (
    <div>
      <p>Typescript test:</p>
      {Object.entries(params)?.map(([key, value]) => (
        <p key={key}>
          <span>{key}</span>: <span>{value}</span>
        </p>
      ))}
    </div>
  );
};

export default AppTs;
```